### PR TITLE
Add data field to loader

### PR DESCRIPTION
**Pull request by kpdecker**
Allows client code to pass data to the application framework at runtime. If
two parameters are provided, the first parameter will be passed verbatim to
the runtime loader. This allows for any code construct as long as the runtime
supports it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,17 @@ Will generate two different chunks, one for the home router and one for the item
 
 Generally a loader is used for simple bootstrapping of an application, along with core libraries.
 
+If two parameters are passed, the first paramter is passed to the loader runtime and maybe used as needed for initialization.
+
+```javascript
+Circus.loader({root: '/foo'}, [
+  './home',
```
Comment (jherr) [inline README.md]: Could you add a little more information about the second argument, the array, and what that's used for.

Comment (divmain) [inline README.md]: That's described a few lines above.

Comment (jherr) [inline README.md]: I still don't get it, but I haven't drilled down into the code.

Comment (divmain) [inline README.md]: The idea here is to tie chunks of code to SPA routes.

At compilation, this plugin will parse any calls to `Circus.loader` (where `Circus` is the lib's namespace), resolve elements in the array in the same way that a `require` statement would, then identify the dependencies in those modules.  Those dependencies are segmented into their own webpack chunks.

At run-time when a route is hit, the hash of routes-to-modules is checked.  If the module (and its deps) associated with the new route is not loaded, it is executed first and then the route is re-triggered.

@kpdecker can you confirm/correct?

File: test/plugins/loader.js [hunk 2]
```diff
@@ -269,6 +269,32 @@ describe('loader plugin', function() {
       done();
     });
   });
+  it.only('should include app data if passed', function(done) {
```
Comment (divmain) [inline test/plugins/loader.js]: `.only`

File: test/plugins/loader.js [hunk 3]
```diff
@@ -269,6 +269,32 @@ describe('loader plugin', function() {
...
+    config = CircusRouter.config(config);
+    config = Circus.config(config);
+
+    webpack(config, function(err, status) {
+      expect(err).to.not.exist;
+      expect(status.compilation.errors).to.be.empty;
+      expect(status.compilation.warnings).to.be.empty;
+
+      var output = fs.readFileSync(outputDir + '/bundle.js').toString();
+      expect(output).to.match(/Circus.loader\(__webpack_require__, \{"data":\{root: '\/root'\},"modules":\{"2":\{"chunk":1\}\},"routes":\{"\/foo":2,"\/bar":2\}\}\);/);
```
Comment (divmain) [inline test/plugins/loader.js]: Why `.to.match`?  All the escapes in the regex make it a bit hard to read.  Does `.to.contain` work here, just passing a string?

**Comment by divmain** [comment]
Code makes sense to me, but I'm not sure I understand the use case.  What sort of run-time data do you envision would be passed from the client code?

**Comment by jherr** [comment]
A root for the backbone router.

File: README.md (hunk 1 above)
Comment (jherr) [inline README.md]: If correct please paste into the README.md because that's a solid explanation right there. Nice job.

Comment (kpdecker) [inline README.md]: @divmain correct, although the run-time behavior that you mention is not implemented in this library and is left to the implementor. We might want to consider publishing our backbone router as a separate project and linking to that if it helps get people started.
@jherr I believe the readme is stating enough of this already while avoiding specifics that are not implemented here but if you have a better way to state this (as a published author :)) then PRs are accepted. I'd recommend reading the docs for the circus project as well as some of Dale's comments are documented there.

**Comment by kpdecker** [comment]
@divmain this is to support the internal PR that I mentioned in the Slack channel. Ping me on Slack if you have questions on that. Long story short, we need to allow applications to pass configuration parameters to the router initialization code.

File: README.md (hunk 1 above)
Comment (jherr) [inline README.md]: Cool, jsdoc works. Let's just get this landed. LGTM.

File: test/plugins/loader.js (hunk 2 above)
Comment (kpdecker) [inline test/plugins/loader.js]: Doh. Removed.

File: test/plugins/loader.js (hunk 3 above)
Comment (kpdecker) [inline test/plugins/loader.js]: `to.match` out of habit/consistency. `to.contain` is a bit easier to read in this case (although both suck :)

Will update this after I get the upstream webpack failure fixed.

**Comment by kpdecker** [comment]
@divmain rebased and made the requested changes. I'll leave this to you for merge or at least final +1.

**Comment by divmain** [comment]
@kpdecker, looks good to me.  Looks like I'm not added as collaborator yet - can you add me, as well as the other circus repos?

**Comment by kpdecker** [comment]
Released in 1.1.0

**Pull request merged**